### PR TITLE
[Resolves #1] Add analyzer starter

### LINKED ISSUE #1: Create analyzer skeleton
See https://pkg.go.dev/golang.org/x/tools/go/analysis

The core idea is to create a package called `analyzer` which exposes a variable `Analyzer` containing the name of the analyzer (returncheck) and an empty run function.

Also, create a `cmd/returncheck` package to expose a singlechecker main function. This allows you to run the analyzer on the CLI. See for example https://github.com/GaijinEntertainment/go-exhaustruct/blob/master/cmd/exhaustruct/main.go

### PR DESCRIPTION
**Pull request by navijation**
Add the analyzer variable as well as the singlechecker. Create `go.mod` to include relevant dependencies and specify Go version 1.22.

### CODE REVIEW

**Comment by navijation** [comment]
I decided to put the analyzer variable definition at the top-level package for ease of discoverability. The implementation logic will live in subpackages.

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The `returncheck.Analyzer` variable follows guidelines in the golang.org/x/tools
 
 # Motivation
 
-In Go, functions that have no side-effects (so-called pure functions) can be called without storing or otherwise using their return values. This is almost always a bug. Other functions have side side-effects but also return values in which the caller should take careful interest. Accidentally disregarding these values _might_ be a bug. A common example is functions that return an updated value rather than mutating the original value.
+In Go, functions that have no side-effects (so-called pure functions) can be called without storing or otherwise using their return values. This is almost always a bug. Other functions have side side-effects but also return values in which the caller should take careful interest. Accidentally disregarding these values _might_ be a bug. A common example of the former case is functions that return an updated value rather than mutating the original value.
```
Comment (Esmeralda-Montas) [inline README.md:18]: Nice summary, it would be good to include a code example. Engineers speak code 🙂

File: returncheck.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+var Analyzer = &analysis.Analyzer{
+	Name:  "returncheck",
+	Doc:   "Go analyzer that requires function return values be used or explicitly ignored with `_`",
+	URL:   "https://github.com/navijation/go-returncheck",
+	Flags: flag.FlagSet{},
+	Run: func(*analysis.Pass) (interface{}, error) {
+		return nil, nil
+	},
+	// TODO: it should actually be possible to run this analyzer in spite of type errors, although
```
Comment (Esmeralda-Montas) [inline returncheck.go:18]: Is there a way to not run this for specific type errors like wrong interface implementations? My thought here is if a method is using an outdated/wrong function signature we may not run the analyzer or it should also surface this error

File: returncheck.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	// certain features like comment directives on function definitions will not work correctly
+	RunDespiteErrors: false,
+	Requires:         []*analysis.Analyzer{inspect.Analyzer},
+	ResultType:       nil,
+	// TODO: facts are necessary to support comment directives on function and struct definitions;
```
Comment (Esmeralda-Montas) [inline returncheck.go:23]: 👍

**Review by Esmeralda-Montas — APPROVED** [review]
non-blocking comments and questions

thanks for getting the ball rolling with this project 🎈

File: returncheck.go (hunk 2 above)
Comment (ImprintNav) [reply returncheck.go:18]: I'm not sure it's possible to see what kinds of errors there were during parsing. But basically my idea here is to try to fallback to the simplest logic, which is:

1. Check to see if an expression statement wraps a call expression
2. If so, try to lookup the return signature of the function using `pass.TypesInfo.TypeOf`
3. If so, check to see if returns any values and report an error

If the second step fails, we can report nothing and move on.

Comment (navijation) [reply returncheck.go:18]: I'm not sure it's possible to see what kinds of errors there were during parsing. But I don't think incorrect interface implementations should affect things. Ultimately it's a question of looking at a single naked function call and looking at the signature of that one function. If the caller forgot to update the underlying type for an interface, it doesn't change the fact that this one call site (presumably a call to an interface) is problematic.

**Pull request merged**